sky/services/vsync/src/org/domokit/vsync/VSyncProviderImpl.java

```diff
@@ -31,17 +33,15 @@ public void onConnectionError(MojoException e) {}
 
     @Override
     public void awaitVSync(final AwaitVSyncResponse callback) {
-        if (mCallback != null) {
-            mPipe.close();
-            return;
-        }
-        mCallback = callback;
+        mCallbacks.add(callback);
         mChoreographer.postFrameCallback(this);
```

Comment:
You should probably only post the frame callback on the choreographer on the first addition to the array. Not that it matters since the callbacks are cleared in `doFrame`. But could save a round trip.